### PR TITLE
[Instrumentation.StackExchangeRedis] Code cleanup

### PR DESCRIPTION
**Pull request by Kielek**
Fixes: N/A.

## Changes

This package is the next candidate to utilize in AutoIntrumentation. I have go through the code to understand how it is working. Please check following improvements:

1. Reference Otel Api by `OpenTelemetryApiPkgVer`
1. Typo fix
1. Cleanup code
    * simplify if condition
    * remove unused default parameter in private method
    * remove unused variable
1. Tests - bump `OpenTelemetry.Extensions.Hosting` to 1.0.0-rc9.4

For significant contributions please make sure you have completed the following items:

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ public static class TracerProviderBuilderExtensions
             TracerProviderBuilder builder,
             IConnectionMultiplexer connection,
             StackExchangeRedisCallsInstrumentationOptions options,
-            Action<StackExchangeRedisCallsInstrumentationOptions> configure = null)
+            Action<StackExchangeRedisCallsInstrumentationOptions> configure)
```
Comment (pellared) [inline src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs:81]: This is a breaking change. The default `null` value is there as often the user is OK with the default `StackExchangeRedisCallsInstrumentationOptions` value (pardon my English).

**Review by pellared — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs (hunk 1 above)
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs:81]: @Kielek You I think should revert this change. This requires a user to always pass in an argument to the extension method.

I think this looks much better 
```csharp
.AddRedisInstrumentation(connection, options)
```

than this:
```csharp
.AddRedisInstrumentation(connection, options, null)
```

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs:81]: This also exposes an issue with publicAPI not running for this project.

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs:81]: @utpilla, it is private method `private static TracerProviderBuilder AddRedisInstrumentation`

All calls are in 
`dotnet
        public static TracerProviderBuilder AddRedisInstrumentation(
            this TracerProviderBuilder builder,
            IConnectionMultiplexer connection = null,
            Action<StackExchangeRedisCallsInstrumentationOptions> configure = null)
`

Public API is fine.

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs:81]: My bad! Never mind 😂
